File: java-example/src/test/java/org/onflow/examples/java/getNodeVersionInfo/GetNodeVersionInfoAccessAPIConnectorTest.java
package org.onflow.examples.java.getNodeVersionInfo;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.onflow.flow.common.test.FlowEmulatorProjectTest;
import org.onflow.flow.common.test.FlowTestClient;
import org.onflow.flow.sdk.FlowAccessApi;
import org.onflow.flow.sdk.FlowNodeVersionInfo;

import static org.junit.jupiter.api.Assertions.*;

@FlowEmulatorProjectTest(flowJsonLocation = "../flow/flow.json")
public class GetNodeVersionInfoAccessAPIConnectorTest {
    @FlowTestClient
    private FlowAccessApi accessAPI;
    private GetNodeVersionInfoAccessAPIConnector nodeVersionInfoConnector;
    @BeforeEach
    public void setup() {
        nodeVersionInfoConnector = new GetNodeVersionInfoAccessAPIConnector(accessAPI);
    }

    @Test
    public void canFetchNodeVersionInfo() {
        FlowNodeVersionInfo nodeVersionInfo = nodeVersionInfoConnector.getNodeVersionInfo();
        assertNotNull(nodeVersionInfo, "Node version info should not be null");
        assertEquals(nodeVersionInfo.getProtocolVersion(), 0);
        assertEquals(nodeVersionInfo.getSporkRootBlockHeight(), 0);
        assertEquals(nodeVersionInfo.getNodeRootBlockHeight(), 0);
        assertNull(nodeVersionInfo.getCompatibleRange());
    }
}

File: kotlin-example/src/test/kotlin/org/onflow/examples/kotlin/getNodeVersionInfo/GetNodeVersionInfoAccessAPIConnectorTest.kt
package org.onflow.examples.kotlin.getNodeVersionInfo

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.onflow.flow.common.test.FlowEmulatorProjectTest
import org.onflow.flow.common.test.FlowTestClient
import org.onflow.flow.sdk.FlowAccessApi
import org.onflow.flow.sdk.FlowChainId
import org.onflow.flow.sdk.FlowNodeVersionInfo

@FlowEmulatorProjectTest(flowJsonLocation = "../flow/flow.json")
internal class GetNodeVersionInfoAccessAPIConnectorTest {
    @FlowTestClient
    lateinit var accessAPI: FlowAccessApi

    private lateinit var nodeVersionInfoConnector: GetNodeVersionInfoAccessAPIConnector

    @BeforeEach
    fun setup() {
        nodeVersionInfoConnector = GetNodeVersionInfoAccessAPIConnector(accessAPI)
    }

    @Test
    fun `Can fetch network parameters`() {
        val nodeVersionInfo: FlowNodeVersionInfo = nodeVersionInfoConnector.getNodeVersionInfo()
        assertNotNull(nodeVersionInfo, "Node version info should not be null")
        assertEquals(nodeVersionInfo.protocolVersion, 0)
        assertEquals(nodeVersionInfo.sporkRootBlockHeight, 0)
        assertEquals(nodeVersionInfo.nodeRootBlockHeight, 0)
        assertEquals(nodeVersionInfo.compatibleRange, null)
    }
}
